Code/Mantid/Framework/PythonInterface/test/python/plugins/algorithms/DNSFlippingRatioCorrTest.py

```diff
@@ -0,0 +1,281 @@
+import unittest
+from testhelpers import run_algorithm
+from mantid.api import AnalysisDataService
+import numpy as np
+import mantid.simpleapi as api
+from mantid.simpleapi import DNSFlippingRatioCorr
+
+
+class DNSFlippingRatioCorrTest(unittest.TestCase):
+    workspaces = []
+    __sf_bkgrws = None
+    __nsf_bkgrws = None
+    __sf_nicrws = None
+    __nsf_nicrws = None
+
+    def _create_fake_workspace(self, wsname, dataY, flipper):
+        """
+        creates DNS workspace with fake data
+        """
+        ndet = 24
+        dataX = np.zeros(2*ndet)
+        dataX.fill(4.2 + 0.00001)
+        dataX[::2] -= 0.000002
+        dataE = np.sqrt(dataY)
+        # create workspace
+        api.CreateWorkspace(OutputWorkspace=wsname, DataX=dataX, DataY=dataY,
+                            DataE=dataE, NSpec=ndet, UnitX="Wavelength")
+        outws = api.mtd[wsname]
+        api.LoadInstrument(outws, InstrumentName='DNS')
```

Comment:
Thank you for recommendation. Since this function is any way subject for refactoring (because I want to make it generic for testing of all DNS algorithms), I will take a look whether it is possible to make it faster. For my unittests is important to have a connection to the real DNS instrument to avoid problems if IDF will be changed.